java/src/jmri/StringIO.java

```diff
@@ -0,0 +1,58 @@
+package jmri;
+
+/**
+ * Represent an string I/O on the layout.
+ * <P>
+ * A StringIO could for example being a display connected to an Arduino
+ * microcomputer that shows train departures of a station.
+ *
+ * @author Daniel Bergqvist Copyright (c) 2018
+ */
+public interface StringIO extends NamedBean {
+
+    /**
+     * Change the commanded state, which results in the relevant command(s)
+     * being sent to the hardware. The exception is thrown if there are problems
+     * communicating with the layout hardware.
+     *
+     * @param value the desired string value
+     * @throws jmri.JmriException general error when setting the state fails
+     */
+    public void setCommandedStringValue(String value) throws JmriException;
```

Comment:
Can we mark all the strings in the interface as `@Nonnull`?  I.e. they can be the empty string "", but they'll never be null?

On the other hand, perhaps `null` represents the `INCONSISTENT` state of "nobody has set a value yet".

Either way is probably fine, but it should be made explicit.

Reply:
Yes. Fixed.